### ISSUE TITLE
make use of linalg within Shogun

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#2747]
This task is to make Shogun's internal make use of the new [linalg framework](https://github.com/shogun-toolbox/shogun/wiki/README_linalg)

It is pretty simple: Take a method that you want to replace, say some dot product. Then go through Shogun's code and search for appearances of dot products. Those could take a lot of different forms

 * manual loops
 * lapack calls (look up the dot product call in lapack)
 * eigen3 calls (easier to find, search for eigen3 in the code)

For example, linear solves are much easier to find as you can search for eigen3/lapack's method for that directly.

**Comment by karlnapf** [comment]
Useful for the linalg GSoC project. https://github.com/shogun-toolbox/shogun/wiki/GSoC_2015_project_linalg

**Comment by sanuj** [comment]
Are we using c++11 everywhere in shogun then?

**Comment by SunilMahendrakar** [comment]
@karlnapf I have already started on this with scale https://github.com/shogun-toolbox/shogun/issues/2717 pull https://github.com/shogun-toolbox/shogun/pull/2744. I will add more soon.

**Comment by karlnapf** [comment]
@sanuj it is not yet done. @lambday will guide this and merge the requirement patch too ;)

**Comment by ajaybhat** [comment]
@karlnapf & @SunilMahendrakar could i know the status of this? Maybe I could pick up where you left off.

**Comment by ckousik** [comment]
https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/distributions/Gaussian.cpp#L318
@karlnapf Shouldn't the LAPACK call be scrapped in favor of the linalg call instead of the conditional compilation?

**Comment by karlnapf** [comment]
absolutely! I think most of linalg that is needed is there, so that should be easy. Make sure to send a few minimal unit tests before modifying the code ...

**Comment by karlnapf** [comment]
There are still many loops who wait to be replaced with linalg calls
